lib/internal/mime.js

```diff
@@ -0,0 +1,339 @@
+        StringPrototypeSlice(insideMatch[0], 0, -1) :
+        insideMatch[0];
+      // Unescape '\' quoted characters
+      parameterValue = StringPrototypeReplace(inside, QUOTED_CHARACTER, '$1');
+      // If we did have an unmatched '\' add it back to the end
+      if (insideMatch[1]) parameterValue += '\\';
+    } else {
+      // Handle the normal parameter value form
+      const valueEnd = StringPrototypeIndexOf(str, SEMICOLON, position);
+      const rawValue = valueEnd === -1 ?
+        StringPrototypeSlice(str, position) :
+        StringPrototypeSlice(str, position, valueEnd);
+      position += rawValue.length;
+      const trimmedValue = StringPrototypeReplace(rawValue, ENDING_WHITESPACE, '');
+      // Ignore parameters without values
+      if (trimmedValue === '') continue;
+      parameterValue = trimmedValue;
+    }
+    if (parameterString !== '' &&
+      StringPrototypeSearch(parameterString, NotHTTPTokenCodePoint) === -1 &&
+      StringPrototypeSearch(parameterValue, NotHTTPQuotedStringCodePoint) === -1 &&
+      paramsMap.has(parameterString) === false) {
+      paramsMap.set(parameterString, parameterValue);
+    }
+    position++;
+  }
+  return paramsMap;
+}
+
+const execPattern = (pattern, value) => {
+  pattern.lastIndex = 0;
+  return RegExpPrototypeExec(pattern, value);
+};
+
+const QUOTE_OR_SOLIDUS = hardenRegExp(/["\\]/g);
+const encode = (value) => {
+  if (value.length === 0) return '""';
+  NotHTTPTokenCodePoint.lastIndex = 0;
+  const encode = StringPrototypeSearch(value, NotHTTPTokenCodePoint) !== -1;
+  if (!encode) return value;
+  const escaped = StringPrototypeReplace(value, QUOTE_OR_SOLIDUS, '\\$&');
+  return `"${escaped}"`;
+};
+
+const MIMEStringify = (type, subtype, parameters) => {
+  let ret = `${type}/${subtype}`;
+  const paramStr = MIMEParamsStringify(parameters);
+  if (paramStr.length) ret += `;${paramStr}`;
+  return ret;
+};
+
+const MIMEParamsStringify = (parameters) => {
+  let ret = '';
+  const entries = MIMEParamsData(parameters).entries();
+  let keyValuePair, done;
+  // Using this to avoid prototype pollution on Map iterators
+  while ({ value: keyValuePair, done } =
+        MapIteratorPrototypeNext(entries)) {
+    if (done) break;
+    const [key, value] = keyValuePair;
+    const encoded = encode(value);
+    // Ensure they are separated
+    if (ret.length) ret += ';';
+    ret += `${key}=${encoded}`;
+  }
+  return ret;
+};
+
+class MIMEParams {
+  #data;
+  constructor(init = '') {
+    this.#data = parseParametersString(`${init}`);
+  }
+
+  delete(name) {
+    this.#data.delete(name);
+  }
+
+  get(name) {
+    const data = this.#data;
+    if (data.has(name)) {
+      return data.get(name);
+    }
+    return null;
+  }
+
+  has(name) {
+    return this.#data.has(name);
+  }
+
+  set(name, value) {
+    const data = this.#data;
+    NotHTTPTokenCodePoint.lastIndex = 0;
+    name = `${name}`;
+    value = `${value}`;
```

Comment:
According to the doc this method accepts `string`s as arguments, I think throwing with `validateString` on incorrect input would be better. (`validateString(name, 'name')` and `validateString(value, 'value')`)

Reply:
Per WebIDL types act as conversions not as assertions and this is based upon WebIDL from https://bmeck.github.io/node-proposal-mime-api/ in this case we are going through https://heycam.github.io/webidl/#es-DOMString . We can update the docs if desired but Node's docs do not establish typing rules clearly in this scenario to my knowledge.